distributed/protocol/tests/test_torch.py

```diff
@@ -18,10 +18,20 @@ def test_grad():
     x = np.arange(10)
     t = torch.Tensor(x)
     t.grad = torch.zeros_like(t) + 1
+    t.requires_grad = True
 
     t2 = deserialize(*serialize(t))
-    assert (t2.numpy() == x).all()
+
+    assert (t2.detach().numpy() == x).all()
     assert (t2.grad.numpy() == 1).all()
+    assert (t2.requires_grad is True)
+
+    t.requires_grad = False
+    t3 = deserialize(*serialize(t))
+
```

Comment:
I would test that serialization and de-serialization does not modify `t.requires_grad`. Adding this line below the serialization would test that:

``` python
assert t.requires_grad == requires_grad
```

Reply:
Done, too.